src/control/tools/EditSelectionContents.cpp

```diff
@@ -232,6 +233,38 @@ auto EditSelectionContents::setFont(XojFont& font) -> UndoAction* {
     return nullptr;
 }
 
+/**
+ * Set the line style of all strokes, return an undo action
+ * (Or nullptr if nothing done)
+ */
+auto EditSelectionContents::setLineStyle(LineStyle style) -> UndoAction* {
+    auto* undo = new LineStyleUndoAction(this->sourcePage, this->sourceLayer);
+
+    bool found = false;
+
+    for (Element* e: this->selected) {
+        if (e->getType() == ELEMENT_STROKE) {
+            auto s = dynamic_cast<Stroke*>(e);
```

Comment:
If you are sure this Element is a Stroke, use `static_cast`. If not, check if `s` is not nullptr. 

```suggestion
            auto s = static_cast<Stroke*>(e);
```
```suggestion
       if (auto s = dynamic_cast<Stroke*>(e); e->getType() == ELEMENT_STROKE && s) {
            
```